repos/spack_repo/builtin/packages/upcxx/package.py

```diff
@@ -112,27 +81,18 @@ class Upcxx(Package, CudaPackage, ROCmPackage):
         default=False,
         description="Enables UPC++ support for the CUDA memory kind on NVIDIA GPUs.\n"
         + "NOTE: Requires CUDA Driver library be present on the build system",
-        when="@2019.3.0:",
-    )
-    conflicts(
-        "+cuda", when="@:2019.2", msg="UPC++ version 2019.3.0 or newer required for CUDA support"
     )
 
     variant(
         "rocm",
         default=False,
         description="Enables UPC++ support for the ROCm/HIP memory kind on AMD GPUs",
-        when="@2022.3.0:",
-    )
-    conflicts(
-        "+rocm", when="@:2022.2", msg="UPC++ version 2022.3.0 or newer required for ROCm support"
     )
 
     variant(
         "level_zero",
         default=False,
         description="Enables UPC++ support for the Level Zero memory kind on Intel GPUs",
-        when="@2023.3.0:",
```

Comment:
This patch hunk is incorrect, because version 2022.3.0 is still present in the list above.